packages/installer.vm/tools/chocolateyinstall.ps1

```diff
@@ -0,0 +1,222 @@
+    $logPath = Join-Path ${Env:VM_COMMON_DIR} "log.txt"
+    if ((Test-Path $failedPackages)) {
+        VM-Write-Log "ERROR" "For each failed package, you may attempt a manual install via: choco install -y <package_name>"
+        VM-Write-Log "ERROR" "Failed package list saved to: $failedPackages"
+        VM-Write-Log "ERROR" "Please check the following logs for additional errors:"
+        VM-Write-Log "ERROR" "`t$logPath (this file)"
+        VM-Write-Log "ERROR" "`t%PROGRAMDATA%\chocolatey\logs\chocolatey.log"
+        VM-Write-Log "ERROR" "`t%LOCALAPPDATA%\Boxstarter\boxstarter.log"
+    }
+
+    # Display installer log if available
+    if ((Test-Path $logPath)) {
+        Write-Host "[-] Please check the following logs for any errors:" -ForegroundColor Yellow
+        Write-Host "`t[-] $logPath" -ForegroundColor Yellow
+        Write-Host "`t[-] %PROGRAMDATA%\chocolatey\logs\chocolatey.log" -ForegroundColor Yellow
+        Write-Host "`t[-] %LOCALAPPDATA%\Boxstarter\boxstarter.log" -ForegroundColor Yellow
+    }
+
+    # Let users know installation is complete by setting background, playing win sound, and display message box
+    # Set background
+    Set-ItemProperty 'HKCU:\Control Panel\Colors' -Name Background -Value "0 0 0" -Force | Out-Null
+    $backgroundImage = "${Env:VM_COMMON_DIR}\background.png"
+    if ((Test-Path $backgroundImage)) {
+        # Center: 0, Stretch: 2, Fit:6, Fill: 10, Span: 22
+        New-ItemProperty -Path "HKCU:\Control Panel\Desktop" -Name WallpaperStyle -PropertyType String -Value 0 -Force | Out-Null
+        New-ItemProperty -Path "HKCU:\Control Panel\Desktop" -Name TileWallpaper -PropertyType String -Value 0 -Force | Out-Null
+        Add-Type -TypeDefinition @"
+using System;
+using System.Runtime.InteropServices;
+
+public class VMBackground
+{
+    [DllImport("User32.dll",CharSet=CharSet.Unicode)]
+    public static extern int SystemParametersInfo (Int32 uAction, Int32 uParam, String lpvParam, Int32 fuWinIni);
+    [DllImport("User32.dll",CharSet=CharSet.Unicode)]
+    public static extern bool SetSysColors(int cElements, int[] lpaElements, int[] lpaRgbValues);
+}
+"@
+        [VMBackground]::SystemParametersInfo(20, 0, $backgroundImage, 3)
+        [VMBackground]::SetSysColors(1, @(1), @(0x000000))
+    }
+
+    # Play sound
+    $playWav = New-Object System.Media.SoundPlayer
+    $playWav.SoundLocation = 'https://www.winhistory.de/more/winstart/down/owin31.wav'
+    $playWav.PlaySync()
+
+    # Show dialog that install has been complete
+    Add-Type -AssemblyName System.Windows.Forms
+    Add-Type -AssemblyName System.Drawing
+
+    # Create form
+    $form = New-Object System.Windows.Forms.Form
+    $form.Text = "$Env:MandiantVM Installation Complete"
+    $form.TopMost = $true
+    $form.StartPosition = [System.Windows.Forms.FormStartPosition]::CenterScreen
+    $form.Icon = New-Object System.Drawing.Icon(Join-Path $Env:VM_COMMON_DIR "vm.ico")
```

Comment:
```suggestion
    $form.Icon = New-Object System.Drawing.Icon(Join-Path $Env:VM_COMMON_DIR "vm.ico" -Resolve)
```

Let's ensure it exists

Reply:
Just remember that the "-Resolve" flag will throw an error if a path doesn't exist, so you may need a try catch if you use it so the script can still continue versus bailing too early. Otherwise you can manually ensure it exists with a test-path